### PR TITLE
Abort the jump if a ship is disabled mid-jump.

### PR DESCRIPTION
**Pull request by Rakete1111**
**Bugfix:** This PR addresses issue #5058 

## Fix Details

Stop the jump sequence when the ship is disabled while the jump animation is playing. This is mainly done for jump drives, but is also applicable for hyperdrives.

For hyperdrives, this can only happen in the very beginning of the jump, while the ship is accelerating in the origin system. While I haven't tested this for hyperdrives (mainly because of how rare/hard to do it is), the result would be that the ship stops the jump with an increase in velocity.

## Testing Done

Tested this on my main save on Korath raiders. Works as intended, the jump drive animation stops and the Korath raider doesn't jump.

### CODE REVIEW

**Comment by Rakete1111** [comment]
Done. FYI I put the check after the acceleration gets nulled since that is exactly what happens a bit further down in the code when landing.

**Comment by tehhowch** [comment]
> Done. FYI I put the check after the acceleration gets nulled since that is exactly what happens a bit further down in the code when landing.

Yep, I figured that! That behavior makes sense for landing, since we need the sprite to always shrink into the planet.

(I accidentally hit edit post on the mobile view and was thinking I hit quote in reply... 🤦‍♂️)

File: source/Ship.cpp [hunk 1]
```diff
@@ -1165,9 +1165,18 @@ void Ship::Move(vector<Visual> &visuals, list<shared_ptr<Flotsam>> &flotsam)
...
+		if(isDisabled)
+		{
+			hyperspaceSystem = nullptr;
+			hyperspaceCount = 0;
+			return;
+		}
+
 		// Don't apply external acceleration while jumping.
 		acceleration = Point();
-
```
Comment (tehhowch) [inline source/Ship.cpp:1170]: Looks like your change ate some whitespace

Comment (Rakete1111) [reply source/Ship.cpp:1170]: oops, fixed :)

**Comment by Amazinite** [comment]
I will say that I mentioned this issue/pr on stream the other day, and the general idea seemed to be to either leave this mechanic as is, or only have the ship be pulled out of jump for jump drives (but not hyperdrives/scram drives).

**Comment by petervdmeer** [comment]
When I open the Twitch link, then I get the error-message; "Sorry. Unless you’ve got a time machine, that content is unavailable.".

My first impression is also that this change is fine. Can you summarize the arguments against this change? (And/Or the arguments to only do this for jump drives?)

**Comment by Amazinite** [comment]
> When I open the Twitch link, then I get the error-message

Yeah, Twitch deletes videos after two weeks. I have the download for it, though.

TLDR:
* This would make capturing a Raider (or other JD-enabled ships) easier, as right now it's not an uncommon occurrence for you to disable such a ship while it's jumping. If that happens, you need to go on an interesting chase to see where the disabled ship went, while with this PR, as mentioned, it'd just make it easier by giving you a window to disable a ship while it can't fight back. In some sense, this would cheapen the usefulness of hit force weapons, as you wouldn't need to prevent a ship from jumping when you can just disable it while it jumps and keep it in-system at the same time.
* Hyperdrives work by creating a "wave" in spacetime that your ship "rides." If you're somehow disabled while jumping, that wave still exists, so could still push you along to your destination.

**Comment by MCOfficer** [comment]
> 
>     * This would make capturing a Raider (or other JD-enabled ships) easier, as right now it's not an uncommon occurrence for you to disable such a ship while it's jumping. If that happens, you need to go on an interesting chase to see where the disabled ship went, while with this PR, as mentioned, it'd just make it easier by giving you a window to disable a ship while it can't fight back. In some sense, this would cheapen the usefulness of hit force weapons, as you wouldn't need to prevent a ship from jumping when you can just disable it while it jumps and keep it in-system at the same time.

Either i suck, or the chances for this happening are very slim.

>     * Hyperdrives work by creating a "wave" in spacetime that your ship "rides." If you're somehow disabled while jumping, that wave still exists, so could still push you along to your destination.

In lore, the explanation still works. Both hyper- and jumpdrive might be charging up before releasing the energy & jumping.

**Comment by petervdmeer** [comment]
I have seen ships getting disabled while they were jumping (and they still jumped away). Somehow I never considered that the ship would still be present and disabled in one of the adjacent systems.
It is indeed a nice gameplay feature if you can still go and search for the disabled ship in neighbouring systems.

The in-lore explanation also works for me.

**Comment by Amazinite** [comment]
> Either i suck, or the chances for this happening are very slim.

It's a fairly common occurrence for me when I'm attempting to board a Raider early game. I also have this happen often when messing around and trying to board Quarg ships.

**Comment by Amazinite** [comment]
I'm not really interested in this change. (See my past explanations for why.) As such, I'm going to close it.
